Sources/AutoLayout/Constrain.swift

```diff
@@ -17,21 +17,20 @@ public final class ConstraintGroup {
     public init() { }
 
     fileprivate var constraints: [NSLayoutConstraint] = [] {
-        willSet { uninstall(constraints) }
-        didSet { install(constraints) }
+        willSet { uninstall() }
```

Comment:
Should we look at `isActive` flag inside the `willSet`/`didSet` and `uninstall`/`install` accordingly? 🤔

Reply:
Hmm the way we have it now, I don't see that much again, unless Im missing something. I left the `uninstall` for when we're replacing a group to `uninstall` the previous from the view, and the `activate` flag will do the operation its supposed to (tbh I don't know the weight of deactivating twice if for example the flag is false). That would make sense if we change the ordering of the calls, but then the is active wouldn't `activate` or `deactivate` any more right?

Reply:
my point was most geared towards the following scenario where we set new constraints: 
1. it only makes sense to uninstall old constraints if `isActive == true` on `willSet`
2. it may make sense to install new constraints if `isActive == true` on `didSet`

Does it make sense?

Reply:
From that scenario yes, it makes sense, but with the `isActive` `set` property as is we'll be activating the same set of constraints twice won't we?

Reply:
yes, I think we have to keep the `isActive` behavior on `set`, and I also don't see any issue in activating/deactivating the constraints twice if we set both `constraints` and `isActive` at once. Do you? 🤔

Reply:
I couldn't find any problem 😅, and if it's ok with you it's ok with me, I thought there might some problem or performance amp with that double operation, sorry for being "chato" 😅

Reply:
I don't foresee any performance problem, as setting/unsetting the flag to the same value twice in quick succession should simply dirty the layout twice with the same value, which will be picked up in the next layout pass.

We should however ensure that when setting either one of the properties the correct behavior is preserved 👍 

And don't worry, you're not doing anything wrong! 😉 IMO these discussions are healthy and only lead to better solutions 💪